src/Model_Hydro/CPU_Hydro/CPU_Shared_DataReconstruction.cpp

```diff
@@ -1006,57 +1024,52 @@ void Hydro_DataReconstruction( const real g_ConVar   [][ CUBE(FLU_NXT) ],
                ddh_R = cc_C - (real)2.*fc_R + cc_R;
 
 //             3-4. limit slope of L&R
-               if ( dh_LL*dh_L < 0.0 ) {
+               if ( dh_LL*dh_L < (real)0.0 ) {
                   if ( SIGN(dd_L) == SIGN(ddh_L)  &&  SIGN(ddh_L) == SIGN(dd_C) ) {
-                     tmp = SIGN(dd_C) * MIN(C_factor*FABS(dd_L), MIN( (real)3.*FABS(ddh_L), C_factor*FABS(dd_C)));
+                     tmp = SIGN(dd_C) * FMIN(C_factor*FABS(dd_L), FMIN((real)3.*FABS(ddh_L), C_factor*FABS(dd_C)));
                   } else {
-                     tmp = 0.0;
+                     tmp = (real)0.0;
                   } // if ( SIGN(dd_L) == SIGN(ddh_L)  &&  SIGN(ddh_L) == SIGN(dd_C) ) ... else ...
                   fc_L  = (real)0.5*(cc_L+cc_C) - tmp/(real)6.0;
                   dh_LL = fc_L - cc_L;
-                  dh_L  = cc_C - fc_L;
```

Comment:
[Bug] You remove the wrong line. You should keep `dh_L` and remove `dh_LL`.

Reply:
I think I did fix this bug at [here](https://github.com/gamer-project/gamer/pull/253/commits/ce63d6eae1799317f47cfe8abb8827e32fab173f). Please check it again, thanks!